### DISCUSSION TITLE
Authentication problem discussion

### DISCUSSION THREAD

**Discussion by CastagnaIT** [CastagnaIT/plugin.video.netflix#1064] [Developers section]
> And also I found a little idea about what is happenning with the authentication thing

@jakermx you can explain all your idea in this thread

**Comment by jakermx** [comment]
ok

**Reply by jakermx** [reply]
Hi there my dear friend, I hope you are fine and doing good!!....

After what I have seen during my sereash on the rea issue of the Timeouts and ConnectionResets, I saw that I would be nice to unify the sessions and  create a single client for the Requets Module...and also try to help Freso tu update the KODI version to  the Library one, I my opinion, I will try to find another solution becuase NF already moved to  TLS 1.3 and the worst thing...HTTP2 with XPlatform... 

This mae me think If currently the urllib and requests API re not fully done, with basic stuffs, what would happen when  NF decide to restrict access  besad on the capabilities of the client???.....

It could happen what happened when  NF did the MSL Migration....the started little by little, then  in a week, 4 major chages wheres impemented. Thats why I said it....it is  cool framework, but...if there is very little solutions and commits  for big time issues....it just tel me, lets be careful....

Dont you think so?

I know this is your project... but I really love what you are doing....and dont want you to be Marce V2...you are being very good project manager...any way, I hope you look forwar and see what could hapen and react before,as you have been....

Salud!

**Reply by jakermx** [reply]
I said  you  could be the next Marcel, because he did  great stuffm  but he is living his live now...because, he deserves it.

Strange thing,,,, the only 2 addons I loked for since My 1st RPI were Netflix and Spotify....and somehow, both  are DRM  consumed bt MSL.... :D 

F$%k Destiny

**Comment by jakermx** [comment]
[OMG......It looks cool by, Scary!!](https://evertpot.com/h2-parallelism/)

**Reply by jakermx** [reply]
> [OMG......It looks cool by, Scary!!](https://evertpot.com/h2-parallelism/)

Didi you see it?, thats how the chunk responses should be  processed,  but...the L4 is still the same...as 1976.

Have you have heard about D4 DB?

**Comment by CastagnaIT** [comment]
> After what I have seen during my sereash on the rea issue of the Timeouts and ConnectionResets, I saw that I would be nice to unify the sessions and create a single client for the Requets Module...

i had planned to join the service servers before the ConnectionError/Reset occours (and only for py3/matrix)
but there is a lot of time needed, and the subdivision into two separate branches (leia/matrix) took time to adapt
and time to reorganize my station.
My intention was to try to unify all the servers by using the server threading mixin
but this change could cause others side effects that need to be check

the problems with neflix _never end_... every month there is something new
so if you worry too much about everything, you can no longer live...

> This mae me think If currently the urllib and requests API re not fully done, with basic stuffs, what would happen when NF decide to restrict access besad on the capabilities of the client???.....

I think this is a common problem for everyone not only here, python packages like urllib are not yet ready for HTTP2 so there is nothing we can do with this,
but the transition to HTTP/2 is log term and i think backward compatibility HTTP1.1 will be maintained for a long time to allow all software to be updated

i saw the transition of netflix to http2 in the about last months of 2020 is a relatively a new thing

> and also try to help Freso tu update the KODI version to the Library one

if you consider it essential to upgrade to a more recent version of Requests/urllib3 you can ping it will surely tell us what to do
the last update is not too old

who is "Marce"?

**Comment by jakermx** [comment]
https://github.com/marcelveldt       <--------------He

**Comment by jakermx** [comment]
He is another Jojo...well another markus

**Comment by Balanga1** [comment]
I'm trying to get Netflix working under Kodi.

I have installed it but am unable to get authentication working. Authentication is described here:-

Login with Authentication key . CastagnaIT/plugin.video.netflix Wiki . GitHub

and described what needs to be done, ie you need to run NFAuthenticationKey

finally managed to download and extract NFAuthenticationKey_Windows_1.1.0.0.zip but can't get it working. Has anyone else had any success?

**Comment by jakermx** [comment]
Hi mate....I am back from dead....well, not so hard. but.... Iot the1st of 3 surgeries to remove the irons of my left hand, it was suppoused that yesterday I would got the 2nd sirgery preop things...but as the holy week , Ihave been reschedule till the next thurs after hollidays

So, I am here for anything I can support....

I am pretty surprised that there is no more Issues than the base (3) and that you are moving forward smothly.... well done my dear friend.

I have read something about the pickle thing.... but it is hard to handle...but Iwas wondering if it  could be handlde as Aync like AJAX....so Lists just get id:key stuff for listing purproses , and a background worker get the details... it could be cached of better....could be stored in a nonvolotile or low TTL DB so, it could be reused.

If you dont like that way....AFAIK the only way to handle it, is based on visual status change, so grabing last , current and next item on list, is the better way to work as Pickle Serialization , that will not be it...it willact as, but will be predictive serialization....

Anyway...

Glad to back and if possible....glad to support in theese days...

Cheers from Mexico
Cesar

**Comment by jakermx** [comment]
But since it is an online service....caching is the better handller, serialization doent looks to be accure to the  needs or platform, since my POV.

As my POV,  it jst could be useful when navigating...but as i said, makiing an AJAX like navigation...will be  faster and even faster if you implement cache and some DB bsed cache...

anyway, it is your addon, it is your call...

while in hospital, I thought and found a way to create  languages files using online translator,  so I  thought, that you can add a Dialog that ask the user if they want the Add On Install Wizard to translate menus and msgs,  from English to the GUI Locate based Language using Google Translate...so just the first time and every update, the Addon will query GT to get it if possible,if not....a msgwill tell the user  it failed....there are some python based translators or we can do it fully legal and right, registering a key ...
any way...just thinking....

**Comment by CastagnaIT** [comment]
Hi, i wish you stabilization over time it must not be very easy your situation

>  but it is hard to handle...but Iwas wondering if it could be handlde as Aync like AJAX....so Lists just get id:key stuff for listing purproses , and a background worker get the details... it could be cached of better....could be stored in a nonvolotile or low TTL DB so, it could be reused.

> As my POV, it jst could be useful when navigating...but as i said, makiing an AJAX like navigation...will be faster and even faster if you implement cache and some DB bsed cache...

i do not know AJAX, but if you have to make comparisons you have to specify python technology say other languages is not really helpfull..
i really don't understand why you say it is difficult to handle is fully automatized, you have to handle ListItemW equal to ListItem, and it is also recognized by Kodistubs
this shortens the code needed, remove uneeded converter loops, and make really easy the part to create the ListItems

> But since it is an online service....caching is the better handller, serialization doent looks to be accure to the needs or platform, since my POV.

what you are saying? this sentence does not make sense...

**Reply by jakermx** [reply]
Hi there mate

Serialization is used to handle data received from a remote host as object, so it could be used later and could be called easily on its source form. Also it is used when data is need to be  used and handled by other Programming lenguagues, so  source data could persists among all handlers...

AJAX, is a way that Web Pages handles data async, so...what we see  is not all the data  avaiable, till we trigger some elemeent event....that way, pages can grab some data to show, while all data is being loaded....

NF Example...we can navigate and see thumbs, but as we see small thumbs.. title related data isbeingloadad in background, so it feels nice and fast, but itis loaded on demand theuser call for it.

About your last comment...I meant, since itis using AJAX, as it is an asyn online service, serialization doesnt make sense to use, till the client walks among several programming lenguages...got me?

Some thing like MSFT Entity Framework on a  cache based systems, maes more sense to me....so grabbed data are stored in a NV DB, then call be called as object withing any framework  plattform  , otherwise, caching...is good enough

**Comment by jakermx** [comment]
as we both have ran the AddOn on Debug Mode several times...you might noticed that, getting navigation data,is pretty fast, but, it sloows itdown, grabbing the art and pictures thing....so....What I think you want to do is...get , proc, store and show a list of items fast and serialize theitem  detailed data when an list item is invoked, so it couldbe handled as it was  requested, without being reuqested again....

**Comment by amyren1966** [comment]
I had a lot of trouble login in to netflix, bit finally resolved it now.

I run it on Zgemma enigma2 tuner, OpenVix image. It have Kodi Leia 18.8. It always gave me worg password when using the email/pass method. I got the NFAuthention key generated (after some struggle)
Tip: If you fail to get the chome path accepted, remember to use double // for all / in the file path.

I did have Netflix installed from before and whatever I did, I got the key error when using the NFA key. Tried updating the netflix video plugin but nothing helped. 
Then I found the .kodi folder on the HDD, and deleted the folder.
This gave me a fresh start. Now I just installed the correct repo for Kodi 18, and installed the plugin version 1.12.4. This time I used the NFA authention key and finally succeded to get logged in.

Edit: This was a short joy.. It turns out it doesnt play any content. After selcting a video it shows the spinner for a second, and then nothing more happens.

Edit 2: Got it working again. That old kodi folder I deleted. I restored it from a backup and deleted the netflix folder, and then reinstalled the plugin and re-authentiated with the NFA method. Now it plays the videos. But it tells me there is a newer version of widevine CDM available every time I start a video. I just press Cancel instead of updating. 
In my previois attempt of reinstalling I also had to download the widewine, and then it failed to play any content. Could it be something with the new widevine version that make the playback fail?

**Comment by kaynemo** [comment]
I can’t for the life of me get the key to work. I generate it fine, entering login and password into the browser etc. but the add on returns the error: webparcing error: email field is missing . Please help !!!!

**Comment by CastagnaIT** [comment]
i do not know Zgemma enigma2 device/system i cannot help i can say only that current last widevine version (at today) is broken on ARM devices, in any case i dont provide support to Kodi 18 anymore, suggest if possible to switch to Kodi 19

>  can’t for the life of me get the key to work. I generate it fine, entering login and password into the browser etc. but the add on returns the error: webparcing error: email field is missing . Please help !!!!

You do not have specified what version NFA used, ensure you are using last version,
at least to me NFA works as expected, so could be a temporary fails,
or could also be that ntfx are changing parts of the website and ATM is not possible do nothing

**Comment by kaynemo** [comment]
NFAuthentication Key for Linux/MacOS (Version 1.1.4)

This is what I am using

I have tried everything, including wiping the addon completely. The key generates just fine, but when applied to login sequence within addon it says webparcing error: email field not found

I am at a loss

**Reply by kaynemo** [reply]
Would it be too much to ask someone to try out the current situation with NFAuthentication key login via netflix addon? So, in case it is successful I'd know it is my sole problem and not a universal glitch?

**Reply by CastagnaIT** [reply]
at least on Windows works,
i have not tried under linux yet, i will try in next days

**Reply by kaynemo** [reply]
Thank you. And by "windows" you mean kodi running on windows machine? I got kodi on Libreelec Leia

**Reply by CastagnaIT** [reply]
yes in my case Windows 10, you can use NFA to create the key wherever you want

**Reply by kaynemo** [reply]
I understand that and the key was created on a MacOs machine, it seems to me the problem is not with the key, but with the addon, that was what I am trying to get sorted out. I don't have access to window machines, only macs and linuxes, so I can't try that out, but I will consider it. The first time I did this, it was months ago it worked from the same machine flawlessly, I needed to switch an account and now I can't get the thing working.

**Reply by kaynemo** [reply]
BTW lets say I do find a windows machine, does the key expire with time?

**Reply by CastagnaIT** [reply]
@kaynemo please update with this:
[plugin.video.netflix_1.18.4+matrix.1_20220315.zip](https://github.com/CastagnaIT/plugin.video.netflix/files/8254640/plugin.video.netflix_1.18.4%2Bmatrix.1_20220315.zip)
after that try login with key, and provide me a new kodi log, so i can see better what happens

**Reply by kaynemo** [reply]
thanks will do this in the morning... one more stupid question, could it be that pycryptodomex is a must be installed on the kodi box for the key to work and also where is the location of the kodi log on linux? Thanks you for your time and effort, you've been wonderful!

**Reply by kaynemo** [reply]
UPD: the file you offered failed to install "dependancies not met". Looking at the filename, did I forget to mention I am using Libreelec Leia?

**Reply by kaynemo** [reply]
UPD. Generated a key on a Windows machine, still no go. Here is the log. Perhaps you can figure out what is happening. I really appreciate all the time you invest in my problem.
[kodi.log](https://github.com/CastagnaIT/plugin.video.netflix/files/8309192/kodi.log)

**Reply by kaynemo** [reply]
@CastagnaIT  Ummm, sorry to bump the thread. But any thoughts on my case? Thank you.

**Reply by kaynemo** [reply]
@CastagnaIT Hmmm... no thoughts at all?

**Reply by kaynemo** [reply]
OK. This is turning into a monologue ))) But no matter. I've found another test addon from CastagnaIT on the forums that was made for Leia specifically and installed it - lo and behold, the autentification (even with email/password) worked, i updated it through the repo and it seems to still work, it had an error related to missing "utf-8", discussed elsewhere on internet, easily fixed by editing utls.py file.

**Reply by CastagnaIT** [reply]
Sorry for the delay but no more support for Kodi 18

**Reply by kaynemo** [reply]
> Sorry for the delay but no more support for Kodi 18

No problem. As long as you keep on making the sweet software, I can bear getting my hands dirty trying to get my old box running it. Sad to hear about the end of support for Leia, I am a little reluctant to jump to Libreelec 10 with Matrix as it seems there are several problems unresolved yet that may directly impact my set up. But, I guess, I will have to make that jump sooner or later anyway.
